### ISSUE TITLE
Add .info to proxy payload

### ISSUE THREAD

**Issue by michalpristas** [gomods/athens#131]
At the moment when we call upload endpoint on proxy,
payload contains module (.mod) and and zip (.zip). only thing missing is .info

This task involves
- adding info to payload (pkg/payload/upload)
- modifying all .Save methods of all storages so far in place (incl interface pkg/storage/Saver) to accept and store Info bits 

Follow conventions used for zip

Note: at the moment info is consturcted in this way:
```go
RevInfo: storage.RevInfo{
	Version: result.Version,
	Name:    result.Version,
	Short:   result.Version,
	Time:    time.Now(),
},
```

Which serves the purpose but might contain something else as .info file produced by vgo

**Comment by robjloranger** [comment]
I think I've got this, I'll be back if I run into trouble.

**Comment by robjloranger** [comment]
Just waiting on another issue before confirming this as working, #139.

In the meantime, what is the `.info` file supposed to be? Just a hidden config file or something more? i.e. `{go,mod,source}.info` There are a few places where the file name is used but I'm not sure what it should be so I'm just putting `.info` for now.

**Comment by marpio** [comment]
@robjloranger it's JSON-formatted metadata about a version:
```
type RevInfo struct {
	Version string    // version string
	Name    string    // complete ID in underlying repository
	Short   string    // shortened ID, for use in pseudo-version
	Time    time.Time // commit time
}
```
You can read about it here: [https://research.swtch.com/vgo-module](url) in the **Download Protocol** section.

**Comment by michalpristas** [comment]
@robjloranger for uploading info. you do not need to know its content. please treat it as a blob storage.
once this task is done. then generating RevInfo might be replaced by reading info file.

for missing configuration please take a look at travis.yml file. there are bunch of env variables which might be of use

**Comment by michalpristas** [comment]
then on each storage getter.go we do this: https://github.com/gomods/athens/blob/8ceda38ad702b86ae9cb45090273e2e22fb14cbe/pkg/storage/fs/getter.go#L26

```go
return &storage.Version{
		RevInfo: storage.RevInfo{
			Version: version,
			Name:    version,
			Short:   version,
			Time:    time.Now(),
		},...
```

This needs to be replaced by parsed .info file content.
